modules/auxiliary/admin/http/ibm_drm_download.rb

```diff
@@ -0,0 +1,204 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'zip'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'IBM Data Risk Manager Arbitrary File Download',
+        'Description' => %q{
+          IBM Data Risk Manager (IDRM) contains two vulnerabilities that can be chained by
+          an unauthenticated attacker to download arbitrary files off the system.
+          The first is an unauthenticated bypass, followed by a path traversal.
+          This module exploits both vulnerabilities, giving an attacker the ability to download (non-root) files.
+          A downloaded file is zipped, and this module also unzips it before storing it in the database.
+          By default this module downloads Tomcat's application.properties files, which contains the
+          database password, amongst other sensitive data.
+          At the time of disclosure, this is a 0 day. Versions 2.0.3 and 2.0.2 are confirmed to be
+          affected, and the latest 2.0.6 is most likely affected too. Version 2.0.1 is not vulnerable.
+        },
+        'Author' =>
+          [
+            'Pedro Ribeiro <pedrib[at]gmail.com>' # Vulnerability discovery and Metasploit module
+          ],
+        'License' => MSF_LICENSE,
+        'References' =>
+          [
+            [ 'URL', 'https://github.com/pedrib/PoC/blob/master/advisories/IBM/ibm_drm/ibm_drm_rce.md' ],
+            [ 'URL', 'https://seclists.org/fulldisclosure/2020/Apr/33' ],
+            [ 'CVE', '2020-4427' ], # auth bypass
+            [ 'CVE', '2020-4429' ], # insecure default password
+          ],
+        'DisclosureDate' => 'Apr 21 2020'
+      )
```

Comment:
No actions in this module? What does it default to?

Reply:
What do you mean?

Reply:
I'll check it out. No worries!

Reply:
```
msf5 auxiliary(admin/http/ibm_drm_download) > show actions

Auxiliary actions:

   Name  Description
   ----  -----------


msf5 auxiliary(admin/http/ibm_drm_download) > options

Module options (auxiliary/admin/http/ibm_drm_download):

   Name       Current Setting                                                               Required  Description
   ----       ---------------                                                               --------  -----------
   FILEPATH   /home/a3user/Tomcat/webapps/albatross/WEB-INF/classes/application.properties  no        Path of the file to download
   Proxies                                                                                  no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOSTS                                                                                   yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
   RPORT      8443                                                                          yes       The target port (TCP)
   SSL        true                                                                          no        Negotiate SSL/TLS for outgoing connections
   TARGETURI  /                                                                             yes       Default server path
   VHOST                                                                                    no        HTTP server virtual host

msf5 auxiliary(admin/http/ibm_drm_download) >
```

I'll fix it. :)

Reply:
```
msf5 auxiliary(admin/http/ibm_drm_download) > show actions

Auxiliary actions:

   Name      Description
   ----      -----------
   Download  Download arbitrary file


msf5 auxiliary(admin/http/ibm_drm_download) > options

Module options (auxiliary/admin/http/ibm_drm_download):

   Name       Current Setting                                                               Required  Description
   ----       ---------------                                                               --------  -----------
   FILEPATH   /home/a3user/Tomcat/webapps/albatross/WEB-INF/classes/application.properties  no        Path of the file to download
   Proxies                                                                                  no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOSTS                                                                                   yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
   RPORT      8443                                                                          yes       The target port (TCP)
   SSL        true                                                                          no        Negotiate SSL/TLS for outgoing connections
   TARGETURI  /                                                                             yes       Default server path
   VHOST                                                                                    no        HTTP server virtual host


Auxiliary action:

   Name      Description
   ----      -----------
   Download  Download arbitrary file


msf5 auxiliary(admin/http/ibm_drm_download) >
```

Reply:
hmm ok no idea that existed

Reply:
and how do you make that magic happen?

Reply:
Apparently I never committed this change.

Reply:
#13413